src/hotspot/os/linux/os_linux.cpp

```diff
@@ -927,6 +927,14 @@ bool os::create_thread(Thread* thread, ThreadType thr_type,
   }
   assert(is_aligned(stack_size, os::vm_page_size()), "stack_size not aligned");
 
```

Comment:
Small misgiving here about having to scan /proc/meminfo anew for every thread start. The problem is os::large_page_init() bailing out early if UseLargePages=0, right, so we cannot rely on large page initialization?

Reply:
Hi @tstuefe, I made this change. Please take a look. Thanks!